### ISSUE TITLE
Recurring error message when computing paths on contracted graph

### ISSUE THREAD

**Issue by Robinlovelace** [UrbanAnalyst/dodgr#240]
Full reproducible example below, generates the following error:

```
time_dodgr2 <- system.time({
+   paths <- dodgr_paths(graphc, o, d, pairwise = TRUE)
+ })

 *** caught segfault ***
address (nil), cause 'unknown'

Traceback:
 1: rcpp_get_paths_pairwise(graph, vert_map, from_index$index, to_index$index,     heap)
 2: dodgr_paths(graphc, o, d, pairwise = TRUE)
 3: system.time({    paths <- dodgr_paths(graphc, o, d, pairwise = TRUE)})

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace
```

<details>

## Load Inputs

```{r inputs, include=FALSE}
# install dev version of dodgr:
remotes::install_dev("dodgr")
library(dodgr)
library(osmdata)
library(dplyr)
library(sf)
library(tmap)
od_geo <- sf::read_sf("https://github.com/ITSLeeds/routingday/raw/main/input_data/od_data_100_sf.geojson")

leeds_osm <- opq("Leeds, UK") %>%
  add_osm_feature(key = "highway") %>%
  osmdata_sc()
```

# Build graph

```{r nework, include=FALSE}
graph <- weight_streetnet(leeds_osm, "bicycle", turn_penalty = TRUE, left_side = TRUE)
```
# Get Routes

Extracting Origins and Destinations

```{r}
origins <- lwgeom::st_startpoint(od_geo) |> st_as_sf()
origins <- st_coordinates(origins)


destinations <- lwgeom::st_endpoint(od_geo) |> st_as_sf()
destinations <- st_coordinates(destinations)
```

Routing

```{r}
time_dodgr1 <- system.time({
  paths <- dodgr_paths(graph, origins, destinations, pairwise = TRUE)
})
time_dodgr1
time_dodgr1[3]
```

Helper function from UK2GTFS
```{r}
path_to_sf <- function(dp, verts, simplify = FALSE) {
  # Check for emplyr paths
  if (length(dp[[1]]) > 0) {
    path <- verts[match(dp[[1]], verts$id), ]
    path <- matrix(c(path$x, path$y), ncol = 2)
    path <- sf::st_linestring(path)

    if (simplify) {
      path <- sf::st_as_sfc(list(path), crs = 4326)
      path <- sf::st_transform(path, 27700)
      path <- sf::st_simplify(path, 5, preserveTopology = TRUE)
      path <- sf::st_transform(path, 4326)
      path <- path[[1]]
    }
    return(path)
  } else {
    return(NA)
  }
}

paths_to_sf <- function(paths, verts) {
  dp.list <- purrr::map(paths, path_to_sf, verts = verts, .progress = TRUE)
  dp.list <- unname(dp.list)
  dp.list
}
```

```{r}
verts <- dodgr_vertices(graph)
paths2 <- paths_to_sf(paths, verts)
paths2 <- sf::st_as_sfc(paths2, crs = 4326)
```

Plot

```{r}
plot(paths2)
```

# On contracted graph

```{r}
o <- match_pts_to_verts(dodgr_vertices(graph), origins, connected = TRUE)
d <- match_pts_to_verts(dodgr_vertices(graph), destinations, connected = TRUE)
system.time({
  graphc <- dodgr_contract_graph(graph, verts=c(o, d))
})
time_dodgr2 <- system.time({
  paths <- dodgr_paths(graphc, o, d, pairwise = TRUE)
})
```

</details>

Source: https://github.com/ITSLeeds/routingday/pull/15

**Comment by mpadge** [comment]
Thanks @Robinlovelace, being fixed via #241 ...

**Comment by mpadge** [comment]
Fixed
